sdk/iothub/azure-iot-deviceprovisioning/azure/iot/deviceprovisioning/_client.py

```diff
@@ -0,0 +1,200 @@
+    EnrollmentGroupOperations,
+    IndividualEnrollmentOperations,
+)
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from azure.core.credentials import TokenCredential
+
+
+class DeviceProvisioningClient(
+    object
+):  # pylint: disable=client-accepts-api-version-keyword
+    """
+    API for connecting to, and conducting operations on a Device Provisioning Service instance
+
+    :param str endpoint: The HTTP endpoint of the Device Provisioning Service instance
+    :param credential: The credential type used to authenticate with the Device Provisioning Service instance
+    :type credential:
+        ~azure.core.credentials.AzureNamedKeyCredential or
+        ~azure.core.credentials.AzureSasCredential or
+        ~azure.core.credentials.TokenCredential
+    :keyword api_version: The Device Provisioning Service API version to use for requests. Default value is the most
+        recent service version that is compatible with the current SDK. Setting to an older version may result
+        in reduced feature compatibility.
+    :paramtype api_version: str or ApiVersion
+    :ivar individual_enrollment: IndividualEnrollmentOperations operations
+    :vartype individual_enrollment:
+    azure.iot.deviceprovisioningservice.operations.IndividualEnrollmentOperations
+    :ivar enrollment_group: EnrollmentGroupOperations operations
+    :vartype enrollment_group:
+    azure.iot.deviceprovisioningservice.operations.EnrollmentGroupOperations
+    :ivar device_registration_state: DeviceRegistrationStateOperations operations
+    :vartype device_registration_state:
+    azure.iot.deviceprovisioningservice.operations.DeviceRegistrationStateOperations
+    """
+
+    def __init__(
+        self,
+        endpoint: str,
+        credential: Union[
+            "TokenCredential", "AzureNamedKeyCredential", "AzureSasCredential"
+        ],
+        *,
+        api_version: Union[str, ApiVersion] = DEFAULT_VERSION,
+        **kwargs,
+    ) -> None:
+        self._pipeline = self._create_pipeline(
+            credential=credential, base_url=endpoint, **kwargs
+        )
+
+        # Validate endpoint
+        try:
+            if not endpoint.lower().startswith("http"):
+                endpoint = "https://" + endpoint
+        except AttributeError:
+            raise ValueError("Endpoint URL must be a string.")
+        endpoint = endpoint.rstrip("/")
+
+        # Validate api-version
+        try:
+            api_version = ApiVersion(api_version).value
+        except ValueError:
+            raise ValueError(f"Invalid api-version {api_version} specified")
+
+        # Generate base client
+        self._runtime_client = GeneratedDeviceProvisioningClient(
+            credential=credential,  # type: ignore
+            endpoint=endpoint,
+            pipeline=self._pipeline,
+            api_version=api_version,
+            **kwargs,
+        )
+
+        self.individual_enrollment: IndividualEnrollmentOperations = (
```

Comment:
We discussed changing from individual_enrollment -> enrollment. We can do in a following PR if need be.